File: simple-web-server/src/main/java/de/thi/informatik/gdp2/simplewebapp/model/Rooms.java
package de.thi.informatik.gdp2.simplewebapp.model;

import java.util.HashMap;
import java.util.Map;
import java.util.Set;

public class Rooms {
	private Map<String, Room> map;

	public Rooms() {
		this.map = new HashMap<>();
		this.addRoom("Stuff");
		this.addRoom("Funny");
		this.addRoom("Awesome");
	}
	
	public void addRoom(String name) {
		this.map.put(name, new Room(name));
	}

	public Set<String> getRooms() {
		return map.keySet();
	}
	
	public Room getRoom(String name) {
		return this.map.get(name);
	}
}

File: simple-web-server-kotlin/src/main/kotlin/de/thi/informatik/gdp2/simplewebserverkotlin/model/Rooms.kt
package de.thi.informatik.gdp2.simplewebserverkotlin.model

import org.springframework.stereotype.Component

@Component
class Rooms() {
    private val map: MutableMap<String, Room> = HashMap<String, Room>()

    init {
        addRoom("Stuff")
        addRoom("Funny")
        addRoom("Awesome")
    }

    final fun addRoom(name: String) = map.put(name, Room(name))

    fun getRooms(): Set<String> = map.keys

    fun getRoom(name: String): Room? = map[name]
}